src/plugins/ssh-agent/index.ts

```diff
@@ -0,0 +1,146 @@
+/** Copyright © 2024-present P0 Security
+
+This file is part of @p0security/cli
+
+@p0security/cli is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 3 of the License.
+
+@p0security/cli is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with @p0security/cli. If not, see <https://www.gnu.org/licenses/>.
+**/
+import { print2 } from "../../drivers/stdio";
+import { AgentArgs, SshAgentEnv } from "./types";
+import { SpawnOptionsWithoutStdio, spawn } from "node:child_process";
+
+const AUTH_SOCK_MESSAGE = /SSH_AUTH_SOCK=(.+?);/;
+const AGENT_PID_MESSAGE = /SSH_AGENT_PID=(\d+?);/;
+
+/** Spawns a subprocess with given command, args, and options.
+ * May write content to its standard input.
+ * Stdout and stderr of the subprocess is printed to stderr in debug mode.
+ * The returned promise resolves or rejects with the exit code. */
+const asyncSpawn = async (
+  { debug }: AgentArgs,
+  command: string,
+  args?: ReadonlyArray<string>,
+  options?: SpawnOptionsWithoutStdio,
+  writeStdin?: string
+) =>
+  new Promise<number>((resolve, reject) => {
+    const child = spawn(command, args, options);
+
+    if (writeStdin) {
+      if (!child.stdin) return reject("Child process has no stdin");
+      child.stdin.write(writeStdin);
+    }
+
+    child.stdout.on("data", (data) => {
+      if (debug) {
+        print2(data.toString("utf-8"));
+      }
+    });
+
+    child.stderr.on("data", (data) => {
+      if (debug) {
+        print2(data.toString("utf-8"));
+      }
+    });
+
+    child.on("exit", (code) => {
+      if (code !== 0) {
+        return reject(code);
+      }
+      resolve(code);
+    });
+
+    if (writeStdin) {
+      child.stdin?.end();
+    }
+  });
+
+/** Spawns a subprocess with the ssh-agent command.
+ * Detects the auth socket and agent PID from stdout.
+ * Stdout and stderr of the subprocess is printed to stderr in debug mode.
+ * The returned promise resolves with an object that contains the auth socket and agent PID,
+ * or rejects with the contents of stderr. */
+export const sshAgent = async (cmdArgs: AgentArgs) =>
```

Comment:
Do we spawn a new ssh-agent? or are we sharing the existing ssh-agent? If so, are we sure that doing something like `ctrl-c` terminates the agent process as well?

Reply:
Good point! 
`ctrl-d` aka closing the session terminates the agent, however if `ctrl-c` is pressed after the agent is running but before the session is started then the agent remained alive.
I addressed that by listening to termination signals from the main process and killing the agent process in the listener.